saleor/graphql/discount/mutations/promotion/promotion_rule_update.py

```diff
@@ -102,6 +102,8 @@ def clean_input(
             error.code = PromotionRuleUpdateErrorCode.DUPLICATED_INPUT_ITEM.value
             raise ValidationError({"addChannels": error, "removeChannels": error})
         cleaned_input = super().clean_input(info, instance, data, **kwargs)
+        if "gift_ids" in cleaned_input:
+            cleaned_input["gifts"] = cleaned_input.pop("gift_ids")
```

Comment:
Same here

Reply:
Actually `ModelMutation.clean_input` does the trick.